### PR TITLE
Masking: remove flakiness from test

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

`test_custom_4d_attention_mask` was trying to compare the argmax of two sets of logits. The logits should be the same in theory -- one is created through batched inference, the other through properly masked and stacked inputs.

We know from experience that batching introduces tiny fluctuations -- see [this comment](https://github.com/huggingface/transformers/issues/25420#issuecomment-1775317535) explaining why.

As such, the argmax of these two tensors can have different values, resulting in a flaky test. This PR removes the flaky part of the test, the equality check of the argmax. We still check that the two tensors are very similar.

Example of a failing test due to flakiness: https://app.circleci.com/pipelines/github/huggingface/transformers/97709/workflows/f407565a-dd9f-423f-9af2-86f0925cf109/jobs/1294534

### CODE REVIEW

**Comment by amyeroberts** [comment]
Do we know how different they are? Could we check with a tolerance instead? I'm find removing the hard check, but the two outputs being similar seems like a good thing to make sure we're keeping

**Comment by gante** [comment]
@amyeroberts that check already exists 🙌 (expand the diff)

**Review by amyeroberts — APPROVED** [review]
Thanks!
